### PR TITLE
Adding a locale parameter to syslog input

### PR DESCRIPTION
**Pull request by nwknu**
so it can be propagated to the date filter used in the syslog input.
That will allow users of the syslog input to specify the locale that should be used to parse the date, when the locales of the origin and destination are not the same.

Related Jira issue: https://logstash.jira.com/browse/LOGSTASH-1998

### CODE REVIEW

**Comment by gonza8888** [comment]
Hi everyone. I have the same issue.

Is there a solution finally?
locale parameter in input plugin inside .conf file?

File: lib/logstash/inputs/syslog.rb [hunk 1]
```diff
@@ -58,8 +61,10 @@ def register
       "match" => { "message" => "<%{POSINT:priority}>%{SYSLOGLINE}" },
     )
 
+    locale = @config["locale"][0] if @config["locale"] != nil and @config["locale"][0] != nil
```
Comment (suyograo) [inline lib/logstash/inputs/syslog.rb]: Hi @nwknu, we should remove this check as @colinsurprenant pointed out and directly pass it to the date filter if it is not `nil` or empty

**Comment by colinsurprenant** [comment]
just moved this PR to logstash-plugins/logstash-input-syslog#1 
once LGTM'ed we can close here and merge there.

**Comment by colinsurprenant** [comment]
@suyograo can you move your code comment to logstash-plugins ? :P

**Comment by colinsurprenant** [comment]
I completed the work and made some fixes, can we please code review logstash-plugins/logstash-input-syslog#1 ?

**Comment by colinsurprenant** [comment]
we can close this, logstash-plugins/logstash-input-syslog#1 has been merged.

**Pull request closed without merging**